File: c#_SDK/PayHubWS_sample_app/PayHubWS_sample_app/Samples/TransactionReportSample.cs
using PayHubSDK.com.payhub.ws.api;
using PayHubSDK.com.payhub.ws.model;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PayHubWS.Samples
{
    class TransactionReportSample
    {
        public void findReports()
        {
            string url = "https://staging-api.payhub.com/api/v2/";
            string oauth = "bb96358e-2aa8-4c6c-8a2e-901b676e979d";

            Merchant merchant = new Merchant();
            merchant.organization_id = 10127;
            merchant.terminal_id = 215;
            TransactionSearchParameters tsp = new TransactionSearchParameters();
            tsp.AmountFrom = "1";
            tsp.AmountTo = "1002";
            tsp.CardLast4Digits = "4507";
            tsp.BatchIdFrom = "1300";
            tsp.BatchIdTo = "1320";
            tsp.Email = "marrighi";
            tsp.TransactionType = "Sale";
            tsp.ResponseCode = "00";
            tsp.FirstName = "First";
            tsp.LastName = "Cont";
            tsp.PhoneNumber = "(415) 479 1349";
            tsp.TrnDateFrom = "2015-04-01 00:00:00";
            tsp.TrnDateTo = "2015-04-30 00:00:00";
            tsp.CardType = "MasterCard";
            tsp.CardToken = "9999000000001853";
            tsp.Swiped = "true";
            tsp.Source = "3rd Party API";


            TransactionManager transaction = new TransactionManager(url, oauth, merchant);
            List<TransactionReportInformation> tri = transaction.findTransactions(tsp);
            
            Console.Write(tri.ElementAt(0).CustomerName);

        }
    }
}

File: Payhub - SDK/ruby_SDK/Samples/transaction_report_sample.rb
require 'PayHubSDK/com/payhub/ws/extra/include_classes'

wsURL="https://staging-api.payhub.com/api/v2/"
oauth_token = "107d74ab-4a18-4713-88ff-69bd05710086"

merchant = Merchant.new
merchant.organization_id=10127
merchant.terminal_id=215

tsp = TransactionSearchParameters.new
tsp.amountFrom = "1";
tsp.amountTo = "1002";
tsp.cardLast4Digits = "4507";
tsp.batchIdFrom = "1300";
tsp.batchIdTo = "1320";
tsp.email = "marrighi";
tsp.transactionType = "Sale";
tsp.responseCode = "00";
tsp.firstName = "First";
tsp.lastName = "Cont";
tsp.phoneNumber = "(415) 479 1349";
tsp.trnDateFrom = "2015-04-01 00:00:00";
tsp.trnDateTo = "2015-04-30 00:00:00";
tsp.cardType = "MasterCard";
tsp.cardToken = "9999000000001853";
tsp.swiped = "true";
tsp.source = "3rd Party API";
transaction = TransactionManager.new(wsURL,oauth_token,merchant)

result = transaction.findTransactions(tsp)
puts result.at(0).inspect

